File: Backend/Events/Converters/Parties/Digimons/StoredDigievolutionConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Parties.Digimons;

namespace Backend.Events.Converters.Parties.Digimons;

public static class StoredDigievolutionConverter
{
    public static StoredDigievolutionDTO ToDTO(StoredDigievolution storedDigievolution) => new()
    {
        DigievolutionId = storedDigievolution.DigievolutionId,
        Level = storedDigievolution.Level
    };
}

File: Frontend/src/events/converters/parties/digimons/stored-digievolution.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { StoredDigievolutionDTO } from "@/events/dto/parties/digimons/stored-digievolution.dto";
import type { StoredDigievolution } from "@/models";

export class StoredDigievolutionConverter {
  public static convert(storedDigievolutionDto: DeepRequired<StoredDigievolutionDTO>): StoredDigievolution {
    return {
      digievolutionId: storedDigievolutionDto.digievolutionId,
      level: storedDigievolutionDto.level,
    };
  }
}
